src/lib.rs

```diff
@@ -1,36 +1,70 @@
-use std::ptr::null_mut;
-use std::os::raw::c_int;
+mod libxenevtchn;
+extern crate xenevtchn_sys;
+use libxenevtchn::LibXenEvtchn;
+use log::debug;
+use std::convert::TryInto;
 use std::io::Error;
+use std::os::raw::c_int;
+use std::ptr::null_mut;
+use xenevtchn_sys::{evtchn_port_t, xenevtchn_handle, xenevtchn_port_or_error_t};
 
 #[derive(Debug)]
 pub struct XenEventChannel {
-    handle: *mut xenevtchn_sys::xenevtchn_handle,
-    fd: c_int,
-    bind_port: c_int,
+    handle: *mut xenevtchn_handle,
+    pub bind_port: c_int,
```

Comment:
nitpick: Not a fan of making this member public since it should not be able to be modified externally, right? Perhaps create a getter method? Also, I think a rust int should be returned instead of a `c_int`.

Reply:
That is to say, if you decide to implement a getter method you should cast the `c_int` when returning it there. I don't mean that you should change the type of `bind_port` here.